test/smart/test.ml

```diff
@@ -31,30 +28,67 @@ let () = Fmt_tty.setup_std_outputs ~style_renderer:`Ansi_tty ~utf_8:true ()
 let () = Logs.set_reporter (reporter Fmt.stderr)
 let () = Logs.set_level ~all:true (Some Logs.Debug)
 
+(** utils:  *)
+let ( >>? ) x f =
+  let open Lwt.Infix in
+  x >>= function Ok x -> f x | Error err -> Lwt.return_error err
+
+(** conduit-related setup for tests: *)
+
+let fifo = Conduit_lwt.register ~protocol:(module Fifo)
+
+let resolvers_with_fifo ic oc =
+  let resolve _domain_name =
+    Logs.debug (fun m -> m "Call to the local resolver to give named pipes.");
+    Lwt.return_some (ic, oc)
+  in
+  Conduit_lwt.add fifo resolve Conduit.empty
+
+let loopback = Conduit_lwt.register ~protocol:(module Loopback)
+
 let resolvers_with_payloads payloads =
   let resolve _domain_name = Lwt.return_some payloads in
   Conduit_lwt.add loopback resolve Conduit.empty
 
-(* XXX(dinosaure): [tmp] without systemic deletion of directories. *)
+(** Alcotest setup for testing: *)
+let uid = Alcotest.testable Uid.pp Uid.equal
 
-let local_delete_tmp dir = ignore (Bos.OS.Dir.delete ~recurse:true dir)
-let local_tmps = ref Fpath.Set.empty
-let local_tmps_add file = local_tmps := Fpath.Set.add file !local_tmps
-let local_delete_tmps () = Fpath.Set.iter local_delete_tmp !local_tmps
-let local_tmps_exit = ref true
+let git_ref = Alcotest.testable Git.Reference.pp Git.Reference.equal
+
+let ref_contents =
+  Alcotest.testable
+    (Git.Reference.pp_contents ~pp:Uid.pp)
+    (Git.Reference.equal_contents ~equal:Uid.equal)
+
+(** tmp dir management: *)
+
+(** to keep track of directories created by unit tests and clean them up afterwards *)
+module LocalTmps = struct
```

Comment:
We usually use `snake_case` style instead to use `CamlCase` for a module name, can you replace it by `Local_temp`?